includes/class-wc-stripe-webhook-handler.php

```
@@ -746,6 +746,26 @@ public function process_webhook_refund( $notification ) {
 			$order = WC_Stripe_Helper::get_order_by_charge_id( $notification->data->object->id );
 		}
 
+		// Final fallback: locate the order via its payment intent ID. Some orders are missing
+		// the stored charge ID (for example, a lost write during checkout), so neither the
+		// refund ID nor the charge ID can match. The charge object on the notification still
+		// references the payment intent, which is stored on the order at checkout time. When
+		// found this way, back-fill the charge details from the notification so the refund can
+		// be synced and future lookups succeed.
+		if ( ! $order && ! empty( $notification->data->object->payment_intent ) ) {
+			$order = WC_Stripe_Helper::get_order_by_intent_id( $notification->data->object->payment_intent );
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Helper' in method 'process_webhook_refund'. (undefined)

(StaticAccess)